models/siegert_neuron.cpp

```diff
@@ -235,114 +210,62 @@ nest::siegert_neuron::~siegert_neuron()
-  {
-    err = integrand1( upper_bound, &alpha );
-    if ( err > 1e-12 )
-    {
-      upper_bound *= 2.;
-    }
-  }
-
-  gsl_integration_qags( &F, lower_bound, upper_bound, 0.0, 1.49e-8, 1000, gsl_w_, &result, &error );
-
-  // factor 1e3 due to conversion from kHz to Hz, as time constant in ms.
-  return 1e3 * 1. / ( P_.t_ref_ + exp( y_th * y_th ) * result * P_.tau_m_ );
-}
-
-double
-nest::siegert_neuron::siegert2( double theta_shift, double V_reset_shift, double mu, double sigma )
+nest::siegert_neuron::siegert( double mu, double sigma_square )
 {
-  double y_th;
-  y_th = ( theta_shift - mu ) / sigma;
-  double y_r;
-  y_r = ( V_reset_shift - mu ) / sigma;
-
-  double result, error;
-
-  gsl_function F;
-  F.function = &integrand2;
-
-  struct my_params alpha = { y_th, y_r };
-  F.params = &alpha;
+  double sigma = std::sqrt( sigma_square );
 
-  double lower_bound = 0.;
-  double upper_bound = 1.;
-  double err = 1.;
-  while ( err >= 1e-12 )
+  // Catch cases where neurons get no input.
+  // Use (Brunel, 2000) eq. (22) to estimate
+  // firing rate to be ~ 1e-16
+  if ( ( P_.theta_ - mu ) > 6. * sigma )
   {
-    err = integrand2( upper_bound, &alpha );
-    if ( err > 1e-12 )
-    {
-      upper_bound *= 2.;
-    }
+    return 0.;
   }
 
-  gsl_integration_qags( &F, lower_bound, upper_bound, 0.0, 1.49e-8, 1000, gsl_w_, &result, &error );
-
-  // factor 1e3 due to conversion from kHz to Hz, as time constant in ms.
-  return 1e3 * 1. / ( P_.t_ref_ + result * P_.tau_m_ );
-}
-
-double
-nest::siegert_neuron::siegert( double mu, double sigma_square )
-{
-  double sigma = std::sqrt( sigma_square );
-
   // Effective shift of threshold and reset due to colored noise:
-  // alpha = |zeta(1/2)|/sqrt(2) with zeta being the Riemann zeta
+  // alpha = |zeta(1/2)|*sqrt(2) with zeta being the Riemann zeta
   // function (Fourcaud & Brunel, 2002)
   double alpha = 2.0652531522312172;
+  double threshold_shift = alpha / 2. * sqrt( P_.tau_syn_ / P_.tau_m_ );
 
-  double theta_shift = P_.theta_ + sigma * alpha / 2. * sqrt( P_.tau_syn_ / P_.tau_m_ );
-  double V_r_shift = P_.V_reset_ + sigma * alpha / 2. * sqrt( P_.tau_syn_ / P_.tau_m_ );
+  // Scaled and shifted threshold and reset
+  double y_th = ( P_.theta_ - mu ) / sigma + threshold_shift;
+  double y_r = ( P_.V_reset_ - mu ) / sigma + threshold_shift;
 
-  // Catch cases where neurons get no input.
-  // Use (Brunel, 2000) eq. (22) to estimate
-  // firing rate to be ~ 1e-16
-  if ( ( theta_shift - mu ) > 6. * sigma )
+  double integral, result, error;
+  double erfcx_scale = 1.0;
+  gsl_function F;
+  F.function = &erfcx;
+  F.params = &erfcx_scale;
+
+  // Evaluate integral of exp( s^2 ) * ( 1 + erf( s ) ) from y_r to y_th
+  // depending on the sign of y_th and y_r. Uses the scaled complementary
+  // error function erfcx( s ) = exp( s^2 ) * erf( s ).
+  if ( y_r > 0. )
   {
-    return 0.;
+    gsl_integration_qags( &F, y_r, y_th, 0.0, 1.49e-8, 1000, gsl_w_, &result, &error );
```

Comment:
Could you define magic constants as const doubles instead?

Reply:
Thanks, that's much clearer. Better now?